doc/index.rst

```diff
@@ -0,0 +1,599 @@
+   backports this feature.
+
+   .. versionchanged:: 4.1.0
+
+      The ``__final__`` attribute is now set on decorated objects.
+
+.. decorator:: overload
+
+   See :py:func:`typing.overload`.
+
+   Since Python 3.11, this decorator supports runtime introspection
+   through :func:`get_overloads`; ``typing_extensions.overload``
+   backports this feature.
+
+   .. versionchanged:: 4.2.0
+
+      Added to provide support for introspection through :func:`get_overloads` was added.
+
+.. decorator:: override
+
+   See :py:func:`typing.override`. In ``typing`` since 3.12.
+
+   .. versionadded:: 4.4.0
+
+   .. versionchanged:: 4.5.0
+
+      The decorator now sets the ``__override__`` attribute on the decorated
+      object.
+
+.. decorator:: runtime_checkable
+
+   See :py:func:`typing.runtime_checkable`. In ``typing`` since 3.8.
+
+   In Python 3.12, the performance of runtime-checkable protocols was
+   improved, and ``typing_extensions`` backports these performance
+   improvements.
+
+Functions
+~~~~~~~~~
+
+.. function:: assert_never(arg)
+
+   See :py:func:`typing.assert_never`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.1.0
+
+.. function:: assert_type(val, typ)
+
+   See :py:func:`typing.assert_type`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.2.0
+
+.. function:: clear_overloads()
+
+   See :py:func:`typing.clear_overloads`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.2.0
+
+.. function:: get_args(tp)
+
+   See :py:func:`typing.get_args`. In ``typing`` since 3.8.
+
+   This function was changed in 3.9 and 3.10 to deal with :data:`Annotated`
+   and :class:`ParamSpec` correctly; ``typing_extensions`` backports these
+   fixes.
+
+.. function:: get_origin(tp)
+
+   See :py:func:`typing.get_origin`. In ``typing`` since 3.8.
+
+   This function was changed in 3.9 and 3.10 to deal with :data:`Annotated`
+   and :class:`ParamSpec` correctly; ``typing_extensions`` backports these
+   fixes.
+
+.. function:: get_original_bases(cls)
+
+   See :py:func:`types.get_original_bases`. Added to the standard library
+   in Python 3.12.
+
+   This function should always produce correct results when called on classes
+   constructed using features from ``typing_extensions``. However, it may
+   produce incorrect results when called on some :py:class:`NamedTuple` or
+   :py:class:`TypedDict` on Python <=3.11.
+
+   .. versionadded:: 4.6.0
+
+.. function:: get_overloads(func)
+
+   See :py:func:`typing.get_overloads`. In ``typing`` since 3.11.
+
+   Before Python 3.11, this works only with overloads created through
+   :func:`overload`, not with :py:func:`typing.overload`.
+
+   .. versionadded:: 4.2.0
+
+.. function:: get_type_hints(obj, globalns=None, localns=None, include_extras=False)
+
+   See :py:func:`typing.get_type_hints`.
+
+   In Python 3.11, this function was changed to support the new
+   :py:data:`typing.Required`. ``typing_extensions`` backports these
+   fixes.
+
+   .. versionchanged:: 4.1.0
```

Comment:
Presumably we must also have backported the `include_extras` parameter at some point after PEP 593 was implemented in Python 3.9

Reply:
I think that was before we started a changelog. I haven't gone through the git archeology to figure out version changes from that time.

Reply:
Yeah, I tried digging through the git archaeology, and quickly gave up :)